src/pages/ReimbursementAccount/RequestorStep.js

```diff
@@ -89,73 +59,74 @@ class RequestorStep extends React.Component {
     }
 
     /**
-     * Clear the errors associated to keys in values if found and store the new values in the state.
-     *
+     * Get default value from reimbursementAccount or achData
+     * @param {String} fieldName
+     * @param {*} defaultValue
+     * @returns {String}
+     */
+    getDefaultStateForField(fieldName, defaultValue) {
+        return lodashGet(this.props, ['reimbursementAccount', 'achData', fieldName], defaultValue);
+    }
+
+    /**
      * @param {Object} values
+     * @returns {Object}
      */
-    clearErrorsAndSetValues(values) {
-        const renamedFields = {
-            street: 'requestorAddressStreet',
-            city: 'requestorAddressCity',
-            state: 'requestorAddressState',
-            zipCode: 'requestorAddressZipCode',
-        };
-        const newState = {};
-        _.each(values, (value, inputKey) => {
-            const renamedInputKey = lodashGet(renamedFields, inputKey, inputKey);
-            newState[renamedInputKey] = value;
-        });
-        this.setState(newState);
-        BankAccounts.updateReimbursementAccountDraft(newState);
+    validate(values) {
+        const errors = {};
 
-        // Prepare inputKeys for clearing errors
-        const inputKeys = _.keys(values);
+        if (!values.firstName) {
```

Comment:
We need to trim the value first. Right now I can enter a space as my first name and it works. I didn't catch this in the proposal review but I noticed this because previously we used `ValidationUtils.isRequiredFulfilled` here. https://github.com/Expensify/App/blob/368aa352844c5d5aefeb40baba56d626fc2a9f93/src/libs/ValidationUtils.js#L226
We should probably use that method. The same problem occurs for the last name, street, city, and maybe others.


https://user-images.githubusercontent.com/26260477/202531395-627663fc-1946-4ef2-ab0a-895998ef9cec.mov

Reply:
Updated